File: examples/example-64/js/features/step_definitions/webdriver.js
const { Given, When, Then } = require('@cucumber/cucumber');
const assert = require('assert');

Given('the homepage is opened with webdriver', async function () {
    await this.pages.simpleSearch.open();
});

When('{string} is typed into the search box with webdriver', async function (text) {
    await this.pages.simpleSearch.typeSearch(text);
});

When('the {string} button is pressed with webdriver', async function (text) {
    await this.pages.simpleSearch.submitSearch();
});

Then('the results should display {string} with webdriver', async function (text) {
    const resultsText = await this.pages.simpleSearch.getResultsText();
    assert.ok(resultsText.includes(text));
});

File: examples/example-63/ruby/features/step_definitions/webdriver.rb
require 'rspec'

Given('the homepage is opened with webdriver') do
    @simple_search = SimpleSearchPage.new(@driver, 'http://localhost:9999')
    @simple_search.open
end

When('{string} is typed into the search box with webdriver') do |value|
    @simple_search.type_search(value)
end

When('the {string} button is pressed with webdriver') do |btn|
    @simple_search.submit_search
end

Then('the results should display {string} with webdriver') do |results|
    expect(@simple_search.results_text).to eq(results)
end